internal/ent/hooks/documentdata.go

```diff
@@ -0,0 +1,129 @@
+package hooks
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"entgo.io/ent"
+	"entgo.io/ent/dialect/sql"
+	"entgo.io/ent/dialect/sql/sqljson"
+	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/internal/ent/generated/documentdata"
+	"github.com/theopenlane/core/internal/ent/generated/hook"
+	"github.com/theopenlane/core/internal/ent/generated/template"
+	"github.com/theopenlane/core/pkg/enums"
+	"github.com/theopenlane/iam/auth"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	errMissingTemplate                      = errors.New("missing template")
+	errDocInfoDoesNotMatchAuthenticatedUser = errors.New("NDA submission does not match authenticated user")
+	errUserHasAlreadySignedNDA              = errors.New("user has already signed the NDA")
+	errValidationFailed                     = errors.New("validation failed")
+	errMustBeAnonymousUser                  = errors.New("must be an anonymous user")
+	errMissingResponse                      = errors.New("missing response")
+)
+
+// HookDocumentData runs on document data create mutations to set default values that are not provided
+func HookDocumentData() ent.Hook {
```

Comment:
I would rename this since its specific to the NDA type of data, not all DocumentData
```suggestion
func HookDocumentDataNDA() ent.Hook {
```